mad-dog/src/index.js

```diff
@@ -66,55 +63,61 @@ const makeTest = (name, testFn, { numUsers, numCreatorNodes, useZeroIndexedWalle
 }
 
 const testRunner = async tests => {
-  let failedTests = []
+  const failedTests = []
 
   // Run each test
-  for (let { testName, test, numUsers, useZeroIndexedWallet } of tests) {
+  for (const { testName, test, numUsers, useZeroIndexedWallet } of tests) {
     const date = new Date().toISOString()
     const fileLoggerName = `${testName}-${date}`
     const removeLogger = addFileLogger(fileLoggerName)
 
-    logger.info(`Running test [${testName}] at [${date}]`)
-
-    let libsArray = []
-
-    // Add libs with wallet index 0 if flag exists
-    if (useZeroIndexedWallet) {
-      ({ libsArray, numUsers } = await _setLibsArrayWithZeroIndexedWallet(libsArray, numUsers))
-    }
-
-    // Init required libs instances
-    const libsInstances = await Promise.all(
-      _.range(numUsers).map(async i => {
-        const l = new LibsWrapper(i + accountOffset)
-        await l.initLibs()
-        return l
-      })
-    )
-    libsArray = [...libsArray, ...libsInstances]
+    logger.info(`Running test [${testName}]`)
 
-    const executeAll = makeExecuteAll(libsArray)
-    const executeOne = makeExecuteOne(libsArray)
+    const libsInstances = await generateLibsInstances(numUsers, useZeroIndexedWallet)
+    const executeAll = makeExecuteAll(libsInstances)
+    const executeOne = makeExecuteOne(libsInstances)
 
     const { error } = await test({ executeAll, executeOne })
     if (error) {
       const msg = `Failed test [${testName}] with error [${error}]`
       logger.error(msg)
       failedTests.push(msg)
     }
-    logger.info('Removing logger after test execution')
     removeLogger()
   }
 
   if (failedTests.length > 0) throw new Error(`\n${JSON.stringify(failedTests, null, 2)}`)
 }
 
-// This should go away when we have multiple tests.
-//
-// Currently there's a bug where standing up services
-// in the same run as running the tests
-// causes libs init failures, so we stand up services
-// with a separate command.
+async function generateLibsInstances (numUsers, useZeroIndexedWallet = false) {
+  let libsInstances = []
+
+  // Performing certain special actions the 0th indexed wallet (e.g. writing
```

Comment:
are there other actions which require the `zeroIndexedWallet`? I believe there may be in eth-contracts but do we use it anywhere else in mad-dog?

Reply:
the `zeroIndexedWallet` is only used for ipld blacklist tests at the moment